server/test/test.js

```diff
@@ -0,0 +1,218 @@
+          expect(res.body.status).to.eql('success');
+          done();
+        });
+    });
+
+    it('should return code 200 with specified order', (done) => {
+      chai.request(app)
+        .get('/api/v1/orders/2')
+        .end((err, res) => {
+          expect(res).to.have.status(200);
+          expect(res.body).to.have.property('status');
+          expect(res.body).to.have.property('order');
+          expect(res.body.status).to.eql('success');
+          expect(res.body.order.name).to.eql('Andy Smarty');
+          done();
+        });
+    });
+
+    it('should return code 200 with concise success message', (done) => {
+      chai.request(app)
+        .put('/api/v1/orders/2')
+        .send({
+          status: 'accepted',
+        })
+        .end((err, res) => {
+          expect(res).to.have.status(200);
+          expect(res.body).to.have.property('status');
+          expect(res.body).to.have.property('message');
+          expect(res.body.status).to.eql('success');
+          expect(res.body.message).to.eql(
+            'status of order successfully updated'
+          );
+          expect(res.body.order[1].status).to.eql('accepted');
+          done();
+        });
+    });
+  });
+
+  describe('tests for invalid inputs of Fast Food Fast API', () => {
+    describe('tests for invalid inputs in placing order endpoint', () => {
+      it('should return code 400 with error message', (done) => {
+        chai.request(app)
+          .post('/api/v1/orders')
+          .send({
+            name: 'name',
+            order: 'order',
+            status: 'wait',
+          })
+          .end((err, res) => {
+            expect(res).to.have.status(400);
+            expect(res.body.status).to.eql('error');
+            expect(res.body.error).to.eql('the address field is required');
+            done();
+          });
+      });
+      it('should return code 400 with concise error message', (done) => {
+        chai.request(app)
+          .post('/api/v1/orders')
+          .send({
+            name: 'name',
+            order: 'order',
+            address: 'address',
+          })
+          .end((err, res) => {
+            expect(res).to.have.status(400);
+            expect(res.body.status).to.eql('error');
+            expect(res.body.error).to.eql('the status field is required');
+            done();
+          });
+      });
+      it('should return code 400 with simple error message', (done) => {
+        chai.request(app)
+          .post('/api/v1/orders')
+          .send({
+            name: 'name',
+            order: '',
+            status: 'wait',
+            address: 'address',
+          })
+          .end((err, res) => {
+            expect(res).to.have.status(400);
+            expect(res.body.status).to.eql('error');
+            expect(res.body.error).to.eql('order field cannot be blank');
+            done();
+          });
+      });
+      it('should return 400 with error message', (done) => {
```

Comment:
When should this happen. Test descriptions should read `it should do this when that has happened`. Please make sure this applies to all of them